ethexe/common/src/primitives.rs

```diff
@@ -167,6 +167,46 @@ impl CodeAndId {
     }
 }
 
+/// [`NextEraValidators`] represents the all possible states of the next era validators.
+/// The majority of the era time, the next era validators are not known yet.
+/// The state switches to [`NextEraValidators::Elected`] at the election checkpoint by calling `makeElectionAt` in the middleware.
+/// After the commitment is included in a block, the state switches to [`NextEraValidators::Committed`].
+#[derive(Debug, Clone, Default, PartialEq, Eq, Encode, Decode)]
+pub enum NextEraValidators {
```

Comment:
I don't like idea of storing next era validators in DB. Why:
1) Observer must make election - this is ok right now, but should be moved to consensus service in future (consensus call rpc by self)
2) Elected, but not committed validators could be stored in consensus service memory, which is much simpler implementation
3) In propagate_validators_info - is the only one case where you use it. It removes rpc access once per era, but increase code complexity all over gear exe code, I don't think is fare trade